site/content/integrate/apps/_index.md

```diff
@@ -1,33 +1,50 @@
 ---
 title: "Apps (Developers Preview)"
-heading: "Overview"
+heading: "Introduction"
 description: "Apps are lightweight interactive add-ons to Mattermost."
 section: "integrate"
 ---
 
-Apps are lighweight interactive add-ons to Mattermost. Apps can:
+Apps are lightweight, interactive add-ons to Mattermost which use serverless hosting to run without any dedicated infrastructure. Apps can:
 
 - Display interactive, dynamic Modal forms and Message Actions.
-- Attach themselves to locations in the Mattermost UI (e.g. channel bar buttons, post menu, channel menu, commands), and can add their custom /commands with full Autocomplete.
+- Be written in several different languages.
+- Attach themselves to locations in the Mattermost user interface (e.g. channel bar buttons, post menu, channel menu, commands), and can add their custom `/` commands with full Autocomplete.
 - Receive webhooks from Mattermost, and third-parties, and use the Mattermost REST APIs to post messages, etc.
-- Be hosted externally (HTTP), on Mattermost Cloud (AWS Lambda), and soon in self-managed as well as customers' own AWS environments.
-- Be developed in any language*.
+- Be hosted externally (HTTP) and on Mattermost Cloud (AWS Lambda)*.
+- Work on both Mobile and Desktop apps so developers can focus on the functionality of their apps.
+- Be deployed using our Mattermost serverless hosting infrastructure keeping data secure and supporting scalability.
 
 Apps are available as a Developers Preview and we're looking for your feedback! Share constructive feedback in the [Mattermost Apps channel](https://community.mattermost.com/core/channels/mattermost-apps) on our Mattermost community instance.
 
+* Apps will soon be available for self-managed deployments as well as customers' own AWS environments.
+
 Read the [quick start guide]({{< ref  "quick-start-go" >}}) to learn how to write your first app.
 
-# Contents
+## Mattermost serverless app hosting
+
+Mattermost serverless hosting allows for easy app installation from the App Marketplace by a System Admin and uses AWS Lambda serverless technology instead of relying on a physical server. Developers who create apps using a serverless development approach can easily deploy apps securely, efficiently, and at scale in the Mattermost Cloud.
+
+## Developing serverless apps
+
+### Development environment
+
+Visit the [Developer Setup Guide](https://developers.mattermost.com/contribute/server/developer-setup/) for information about setting up your development environment.
+
+### App lifecycle
+
+- [App Lifecycle]({{< ref "lifecycle" >}})
+- godoc: [appsctl](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/cmd/appsctl) - CLI tool used to provision Mattermost Apps in development and production.
```

Comment:
Do we have to use Go doc on external site? Can we render this dynamic content within Sphinx?  @justinegeffen 

(I think it's ugly and drives the user around to different places)

Reply:
Good question. I copied this from the existing content as-is. This repo uses Hugo which I'm sure can handle dynamic content and @hanzei might know how to do that. 

What do you want it to look like in this context? The list of source files but on this page as opposed to a redirect?

Reply:
We did include the godoc sources into hugo for the plugin API and it didn't work out so great. Getting the formatting right is just hard. 3/5 that a godoc link is fine for now. Devs will need to go there anyway if they are writing go code.